transport.go

```diff
@@ -164,13 +186,22 @@ func (t *Transport) refreshToken(ctx context.Context) error {
 	t.appsTransport.BaseURL = t.BaseURL
 	t.appsTransport.Client = t.Client
 	resp, err := t.appsTransport.RoundTrip(req)
+	e := RefreshTokenError{
+		Message:        fmt.Sprintf("received non 2xx response status %q when fetching %v", resp.Status, req.URL),
+		RootCause:      err,
+		InstallationID: t.installationID,
+		Request:        req,
+		Response:       resp,
+	}
 	if err != nil {
-		return fmt.Errorf("could not get access_tokens from GitHub API for installation ID %v: %v", t.installationID, err)
+		e.Message = fmt.Sprintf("could not get access_tokens from GitHub API for installation ID %v: %v", t.installationID, err)
+		return e
 	}
 	defer resp.Body.Close()
```

Comment:
We should probably move this below so that the Response body isn't closed on a non-2xx response.

Reply:
Good catch. Fixed in https://github.com/bradleyfalzon/ghinstallation/pull/45/commits/84225bc666b40a2ac66f9ec3ea5ab24f105e218c